### PR TITLE
Android/feature/activity stream

### PR DESCRIPTION
**Pull request by fihlie**
As stated in #378 👍 Activity stream is a model for representing potential and completed activities using the JSON format and a core functionality for this project. Activity stream coming from backend should be displayed on mobile.

The meeting notes:

https://github.com/bounswe/2021SpringGroup11/wiki/Meeting-20.2---24.12.2021

Related backend issues:
#348
#347
#346
#344
#342
#326

Thanks to all my friends in backend, they have successfully implemented W3 Activity Stream. And in this PR following features are implemented:
- Activity Stream Connection with backend
- Activity Listing on Mobile (you can navigate through bell icon)
- Refresh Button for Activity List
- Some UI improvements to increase responsiveness.

### CODE REVIEW

File: source/android/Renaissance/lib/widget/activity_container.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            Container(
+              padding: EdgeInsets.all(5),
+              child: Text(widget.summary,
+              overflow: TextOverflow.ellipsis,
+              softWrap: true,
+              maxLines: 2,
+              textAlign: TextAlign.start,
+              style: TextStyle(fontSize: 14,)),
+            ),
+            Text(DateFormat('dd-MM-yyyy – kk:mm').format(DateTime.parse(widget.time)),
```
Comment (kursatbakis) [inline source/android/Renaissance/lib/widget/activity_container.dart:41]: put Dots instead of dashes

**Review by kursatbakis — CHANGES_REQUESTED** [review]
our most crucial feature was this one. thank you for handling it.

**Review by Hazerank — APPROVED** [review]
Kürşat already gave feedback that what I was thinking might be better. This was a really crucial part of our application. Thanks for your effort.

**Comment by fihlie** [comment]
Updates have been made based on feedback

**Review by kursatbakis — APPROVED** [review]
thanks for quick fix

**Pull request merged**